kafka/adminapi_test.go

```diff
@@ -0,0 +1,245 @@
+				NumPartitions:     10000,
+				ReplicationFactor: 90,
+				Config:            StringMapToConfigEntries(confStrings, AlterOperationSet),
+			},
+		},
+		nil)
+	if res != nil || err == nil {
+		t.Fatalf("Expected CreateTopics to fail, but got result: %v, err: %v", res, err)
+	}
+	if ctx.Err() != context.DeadlineExceeded {
+		t.Fatalf("Expected DeadlineExceeded, not %v, %v", ctx.Err(), err)
+	}
+
+	// Incorrect input, fail with ErrInvalidArg
+	ctx, cancel = context.WithTimeout(context.Background(), expDuration)
+	defer cancel()
+	res, err = a.CreateTopics(
+		ctx,
+		[]NewTopic{
+			{
+				// Must not specify both ReplicationFactor and ReplicaAssignment
+				Topic:             "mytopic",
+				NumPartitions:     2,
+				ReplicationFactor: 3,
+				ReplicaAssignment: [][]int32{
+					[]int32{1, 2, 3},
+					[]int32{3, 2, 1},
+				},
+			},
+		},
+		nil)
+	if res != nil || err == nil {
+		t.Fatalf("Expected CreateTopics to fail, but got result: %v, err: %v", res, err)
+	}
+	if ctx.Err() != nil {
+		t.Fatalf("Did not expect context to fail: %v", ctx.Err())
+	}
+	if err.(Error).Code() != ErrInvalidArg {
+		t.Fatalf("Expected ErrInvalidArg, not %v", err)
+	}
+
+	// Incorrect input, fail with ErrInvalidArg
+	ctx, cancel = context.WithTimeout(context.Background(), expDuration)
+	defer cancel()
+	res, err = a.CreateTopics(
+		ctx,
+		[]NewTopic{
+			{
+				// ReplicaAssignment must be same length as Numpartitions
+				Topic:         "mytopic",
+				NumPartitions: 7,
+				ReplicaAssignment: [][]int32{
+					[]int32{1, 2, 3},
+					[]int32{3, 2, 1},
+				},
+			},
+		},
+		nil)
+	if res != nil || err == nil {
+		t.Fatalf("Expected CreateTopics to fail, but got result: %v, err: %v", res, err)
+	}
+	if ctx.Err() != nil {
+		t.Fatalf("Did not expect context to fail: %v", ctx.Err())
+	}
+	if err.(Error).Code() != ErrInvalidArg {
+		t.Fatalf("Expected ErrInvalidArg, not %v", err)
+	}
+
+	// Correct input, using options
+	ctx, cancel = context.WithTimeout(context.Background(), expDuration)
+	defer cancel()
+	res, err = a.CreateTopics(
+		ctx,
+		[]NewTopic{
+			{
+				Topic:         "mytopic4",
+				NumPartitions: 9,
+				ReplicaAssignment: [][]int32{
+					[]int32{1},
+					[]int32{2},
+					[]int32{3},
+					[]int32{4},
+					[]int32{1},
+					[]int32{2},
+					[]int32{3},
+					[]int32{4},
+					[]int32{1},
+				},
+				Config: StringMapToConfigEntries(map[string]string{
+					"some.topic.config":  "unchecked",
+					"these.are.verified": "on the broker",
+					"and.this.is":        "just",
+					"a":                  "unit test",
+				}, AlterOperationSet),
+			},
+		},
+		nil)
+	if res != nil || err == nil {
+		t.Fatalf("Expected CreateTopics to fail, but got result: %v, err: %v", res, err)
+	}
+	if ctx.Err() != context.DeadlineExceeded {
+		t.Fatalf("Expected DeadlineExceeded, not %v", ctx.Err())
+	}
+}
+
+// TestAdminAPIs dry-tests most Admin APIs, no broker is needed.
+func TestAdminAPIs(t *testing.T) {
+
+	a, err := NewAdminClient(&ConfigMap{})
```

Comment:
Could DRY up these tests a bit going table-style: https://github.com/golang/go/wiki/TableDrivenTests

Reply:
I'll add that to the Future Fun list :)